### ISSUE TITLE
event not fired when 'active' value changed to true

### ISSUE THREAD

**Issue by morganwu** [PolymerElements/app-route#109]
I was trying to do some processing when route hits by observing 'active' property, but found the observer function was not invoked when value of 'active' property changed from false to true, while the function invoked when it changed from true to false.

**Comment by rictic** [comment]
This appears fixed at head. Take a look at this jsbin: https://jsbin.com/jipeki/1/edit?html,output

**Comment by akc42** [comment]
@rictic The js in you presented above doesn't observe the active property.  That is what is not working correctly.  The jsbin I did for the bug above does.

See http://jsbin.com/cadodu/edit?html,console,output

I presume using Polygit does get you the latest version. 

The reason this is a problem is that setting the active property to false is done with the setter in _resetProperties.  whereas setting it to true is done in _setMulti.  This is such a hack that I can't quite figure out what is wrong.

For the record, I have also some observers that are spread across more than one just the "active" property (I was originally had an observer function like `_activate(active,routeData.name)` and was puzzling why I was getting called multiple times without the values changing, and the above is what resulted from my attempt to simplify what was happening.

**Comment by akc42** [comment]
Why is this not considered a bug?  Without Active Observer working properly I am going to have another mechanism to trigger by ajax request to get data.

**Comment by TimvdLippe** [comment]
I think this issue happens when multiple routes are chained and a toproute becomes inactive, is that true? Pullrequest #125 should fix this behavior.

**Comment by akc42** [comment]
The jsbin I provide above isn't about chained multiple routes.  I am propagating the complete route down to the sub element, pattern matching it for the specific case and trying to directly observe the active property.  It is fired when going to false, the observer is not fired when transitioning to true.

The reason for that is that inside app-route, reset properties is used to set active false.  It uses the proper Polymer setter and the observer is called.  In the transition to true, app-route uses its own attempt at a multiple property update, and the observer isn't fired.  This seems to be related to how Polymer uses an effects object to understand what is related to what, which app-route isn't providing to the calls properly.  Unfortunately, in the limited time I have I haven't got to the bottom of what is missing.

**Comment by akc42** [comment]
I have been doing some more careful debugging of what is going on within app-route when it does `this.__setMulti`.

The bottom line is that `this._notifyPathUp` fires an active-changed event with the event detail containing a path key of value "active" and a value key.  The proper setter for active fires an active changed event where only the detail object has a single key of "value." and NO path key.  It is only in this latter case that the active property observer gets dispatched.

**Comment by akc42** [comment]
As a quick and dirty fix the following update to `this.__setMulti` seems to work.  I am not sure yet whether it would be worth turning this into a pull request.

```
    __setMulti: function(setObj) {
      // HACK(rictic): skirting around 1.0's lack of a setMulti by poking at
      //     internal data structures. I would not advise that you copy this
      //     example.
      //
      //     In the future this will be a feature of Polymer itself.
      //     See: https://github.com/Polymer/polymer/issues/3640
      //
      //     Hacking around with private methods like this is juggling footguns,
      //     and is likely to have unexpected and unsupported rough edges.
      //
      //     Be ye so warned.
      for (var property in setObj) {
        this._propertySetter(property, setObj[property]);
      }

      for (var property in setObj) {
        this._pathEffector(property, this[property]);
//        this._notifyPathUp(property, this[property]);
        var rootName = this._modelForPath(property);
        var dashCaseName = Polymer.CaseMap.camelToDashCase(rootName);
        var eventName = dashCaseName + this._EVENT_CHANGED;
        // use a cached event here (_useCache: true) for efficiency
        this.fire(eventName, {
          value: this[property]
        }, {bubbles: false});


      }

    }

```

**Comment by akc42** [comment]
Whats happening with this.  I provided a pull request some while ago, but nothing has been done with it.  I am still running my own code with the patched up version, but I would like to avoid having to do so

**Comment by aarondrabeck** [comment]
+1

**Comment by cherukumilli** [comment]
@rictic 
Any comments on @akc42 pull request?

**Comment by e111077** [comment]
fixed by #132
